var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -11,8 +11,14 @@ class Chapel(AutotoolsPackage):
        portable, scalable and open-source."""
 
     homepage = "https://chapel-lang.org/"
-    url      = "https://github.com/chapel-lang/chapel/releases/download/1.20.0/chapel-1.20.0.tar.gz"
+    url      = "https://github.com/chapel-lang/chapel/archive/refs/tags/1.24.1.tar.gz"
```

Comment:
The previous `url` is more robust to change, since it's an artifact put there by developers (while the new one is generated automatically by Github and its sha may change over time). Can you revert to the previous url?

Reply:
Sorry for taking so long to reply. 

`https://github.com/chapel-lang/chapel/archive/refs/tags/1.24.1.tar.gz` is the complete repository of tag 1.24.1 release. While `https://github.com/chapel-lang/chapel/releases/download/1.20.0/chapel-1.20.0.tar.gz` is a subset of the former artifact, and its size is smaller in comparison. 

So should I use the latter one, and modified the all sha256sum in the package file? (even those hashes long time ago stand for the complete repository codes tarball, not the small release artifact.)

Reply:
If there are no functional differences, let's add back:
```
https://github.com/chapel-lang/chapel/releases/download/1.20.0/chapel-1.20.0.tar.gz
```
Rationale is that if developers took the time of crafting a source release tarball and upload that, we should prefer it to the one that gets autogenerated.

Reply:
Thanks for your advice! I'll soon calc the sha256sum of those preferred source release tarballs.

Reply:
Fixed it!